app/src/main/java/com/andrew/tasky/agenda/data/agenda/notifications/BootupNotificationBroadcastReceiver.kt

```diff
@@ -0,0 +1,28 @@
+package com.andrew.tasky.agenda.data.agenda.notifications
+
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import com.andrew.tasky.agenda.domain.AgendaRepository
+import dagger.hilt.android.AndroidEntryPoint
+import javax.inject.Inject
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.launch
+
+@AndroidEntryPoint
+class BootupNotificationBroadcastReceiver : BroadcastReceiver() {
+
+    @Inject
+    lateinit var agendaRepository: AgendaRepository
+
+    override fun onReceive(context: Context?, intent: Intent?) {
+        if (intent?.action == Intent.ACTION_BOOT_COMPLETED) {
+            val coroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main)
```

Comment:
The supervisor job isn't really necessary here, since you only launch a single coroutine from this scope. Supervisor job means that if one job fails, the other jobs launched in this scope will continue running, but since you only have a single job here it's redundant